profiles/desktop.py

```diff
@@ -54,6 +54,8 @@ def _prep_function(*args, **kwargs):
 		archinstall.storage['_desktop_profile'] = desktop
 
 	profile = archinstall.Profile(None, desktop)
+	# Set the resolved profile path to the actual desktop environment
+	archinstall.arguments['profile'] = profile
```

Comment:
This is how `ask_user_questions` does it. So should work in theory.

Reply:
ask_user_questions always sets the profile to desktop, so when the config is exported it shows up as
```json
"profile": {
    "path": "/usr/lib/python3.9/site-packages/archinstall/profiles/desktop.py"
}
```

Reply:
> Ah, that's probably an issue in the `Profile().json()` function which overwrites `profile`. Because it should say `awesome` if awesome was chosen for instance. But what I more meant was that the code in `guided` does the same thing as the modification to `--config` is doing:
> 
>
 https://github.com/archlinux/archinstall/blob/1a254700256784c2954fe42e35019b0680236886/examples/guided.py#L204

true, this issue is only there when selecting from prompt, as select_profile returns desktop

>
>
 https://github.com/archlinux/archinstall/blob/1a254700256784c2954fe42e35019b0680236886/examples/guided.py#L202

also, we might need to change
>
>
 https://github.com/archlinux/archinstall/blob/1a254700256784c2954fe42e35019b0680236886/examples/guided.py#L216
if we set profile to a specific DE, otherwise audio install won't be prompted

Reply:
Agreed, this issue is probably a bit complex as we could fix multiple profile support and this would solve it.
But this PR addresses it for now :)

Reply:
@Torxed , I'm not sure if this logic is even necessary, as users might want an audioserver on a minimal system